File: chapter6end/java/sbur-redis/src/main/java/com/thehecklers/sburredis/SburRedisApplication.java
package com.thehecklers.sburredis;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;
import org.springframework.data.redis.connection.RedisConnectionFactory;
import org.springframework.data.redis.core.RedisOperations;
import org.springframework.data.redis.core.RedisTemplate;
import org.springframework.data.redis.serializer.Jackson2JsonRedisSerializer;
import org.springframework.data.redis.serializer.StringRedisSerializer;

@SpringBootApplication
public class SburRedisApplication {
	@Bean
	public RedisOperations<String, Aircraft>
	redisOperations(RedisConnectionFactory factory) {
		Jackson2JsonRedisSerializer<Aircraft> serializer =
				new Jackson2JsonRedisSerializer<>(Aircraft.class);

		RedisTemplate<String, Aircraft> template = new RedisTemplate<>();
		template.setConnectionFactory(factory);
		template.setDefaultSerializer(serializer);
		template.setKeySerializer(new StringRedisSerializer());

		return template;
	}

	public static void main(String[] args) {
		SpringApplication.run(SburRedisApplication.class, args);
	}

}

File: chapter6end/kotlin/sbur-redis/src/main/kotlin/com/thehecklers/sburredis/SburRedisApplication.kt
package com.thehecklers.sburredis

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Bean
import org.springframework.data.redis.connection.RedisConnectionFactory
import org.springframework.data.redis.core.RedisOperations
import org.springframework.data.redis.core.RedisTemplate
import org.springframework.data.redis.serializer.Jackson2JsonRedisSerializer
import org.springframework.data.redis.serializer.StringRedisSerializer

@SpringBootApplication
class SburRedisApplication {
	@Bean
	fun redisOperations(factory: RedisConnectionFactory?): RedisOperations<String, Aircraft>? {
		val serializer = Jackson2JsonRedisSerializer(Aircraft::class.java)

		val template = RedisTemplate<String, Aircraft>()
		template.setConnectionFactory(factory!!)
		template.setDefaultSerializer(serializer)
		template.keySerializer = StringRedisSerializer()

		return template
	}
}

fun main(args: Array<String>) {
	runApplication<SburRedisApplication>(*args)
}
